tests/test_query.py

```diff
@@ -76,58 +78,153 @@ def setUp(self):
         patch.stopall()
 
     def create_and_assert_product_catalog(self):
         """Assert the product catalog was created as expected."""
-        product_catalog = ProductCatalog('mock-name', 'mock-namespace')
-        self.mock_k8s_api.read_namespaced_config_map.assert_called_once_with('mock-name', 'mock-namespace')
+        product_catalog = ProductCatalog('cray-product-catalog', 'mock-namespace')
+        self.mock_k8s_api.list_namespaced_config_map.assert_called_once_with('mock-namespace')
         return product_catalog
 
     def test_create_product_catalog(self):
         """Test creating a simple ProductCatalog."""
+        self.mock_load_config_map_data = patch('cray_product_catalog.query.load_config_map_data').start()
+        self.mock_load_config_map_data.return_value = MOCK_PRODUCTS
         product_catalog = self.create_and_assert_product_catalog()
         expected_names_and_versions = [
             (name, version) for name in ('sat', 'cos') for version in ('2.0.0', '2.0.1')
-        ] + [('other_product', '2.0.0')]
+        ] + [('cpe', '2.0.0'), ('other_product', '2.0.0')]
         actual_names_and_versions = [
             (product.name, product.version) for product in product_catalog.products
         ]
         self.assertEqual(expected_names_and_versions, actual_names_and_versions)
 
     def test_create_product_catalog_invalid_product_data(self):
         """Test creating a ProductCatalog when the product catalog contains invalid YAML."""
-        self.mock_product_catalog_data['sat'] = '\t'
+        self.mock_k8s_api.list_namespaced_config_map.return_value = Mock(items=[MockInvalidYaml()])
         with self.assertRaisesRegex(ProductCatalogError, 'Failed to load ConfigMap data'):
             self.create_and_assert_product_catalog()
 
     def test_create_product_catalog_null_data(self):
         """Test creating a ProductCatalog when the product catalog contains null data."""
-        self.mock_k8s_api.read_namespaced_config_map.return_value = Mock(data=None)
+        self.mock_load_config_map_data = patch('cray_product_catalog.query.load_config_map_data').start()
+        self.mock_load_config_map_data.return_value = []
+        self.mock_k8s_api.list_namespaced_config_map.return_value = Mock(items=[])
         with self.assertRaisesRegex(ProductCatalogError,
-                                    'No data found in mock-namespace/mock-name ConfigMap.'):
+                                    'No ConfigMaps found in mock-namespace namespace.'):
             self.create_and_assert_product_catalog()
 
-    def test_create_product_catalog_invalid_product_schema(self):
-        """Test creating a ProductCatalog when an entry contains valid YAML but does not match schema."""
-        self.mock_k8s_api.read_namespaced_config_map.return_value = Mock(data={
-            'sat': safe_dump({'2.1': {'component_versions': {'docker': 'should be an array'}}})
-        })
+    def check_for_invalid_product_schema(self, prod, ver):
```

Comment:
All schema validation test cases must move the test_schema_validation.py
On in this class.

Reply:
Moved the test cases to test_schema.py